cmd/web/routes.go

```diff
@@ -34,12 +35,14 @@ func (app *application) subscribe(w http.ResponseWriter, r *http.Request) {
 	}
 
 	newEmail := r.PostForm.Get("email")
-	if !isCorrectEmail(newEmail) {
+	v := validator.New()
+	v.Check(validator.IsValidEmail(newEmail), "email", "invalid email")
+	if !v.IsValid() {
 		app.clientError(w, http.StatusUnprocessableEntity)
```

Comment:
StatusUnprocessableEntity error code is not appropriate here because it indicates that email syntax is correct, but could not proccess it. In our case if email is not valid, it means that it is client mistake

https://www.rfc-editor.org/rfc/rfc4918#section-11.2

Reply:
You can return 400 Bad Request

Reply:
Thanks for RFC link, I will replace StatusUnprocessableEntity with StatusBadRequest